*** ORGmod_offline.erl	2006-09-27 22:49:59.000000000 +0200
--- mod_offline.erl	2007-09-13 03:46:22.000000000 +0200
***************
*** 89,94 ****
--- 89,118 ----
      exit(whereis(Proc), stop),
      {wait, Proc}.
  
+ %% patch start (C) 2007 Robert Paciorek, BSD-type licence
+ %% copy or move message from alias JID
+ check_alias_forward(From, To, Packet) ->
+ 	%% obecnie w oparciu o adres konta podstawowego i flage
+ 	%% docelowo w oparciu o AMP (XEP-0079),
+ 	%% jako wariant pośredni można by wdrożyć w oparciu o specjalne wpisy roadster
+ 	#jid{luser = User, lserver = Server} = To,
+ 	Port = open_port({spawn, "PATH_TO_EXTERN_CHECK_SCRIPT"}, [{packet, 2}]),
+ 	Port ! {self(), {command, ["alias\n", User, " ", Server, "\n"]}},
+ 	receive {Port,{data,Data}} -> Data end,
+ 	Port ! {self(), close},
+ 	AliasFlag = string:sub_word(Data, 3),
+ 	if (AliasFlag /= "alias") ->
+ 		ejabberd_router:route(From, jlib:make_jid(string:sub_word(Data, 1), string:sub_word(Data, 2), ""), Packet);
+ 	true ->
+ 		ok
+ 	end,
+ 	if (AliasFlag == "real") ->
+ 		false;
+ 	true ->
+ 		true
+ 	end.
+ %% patch end
+ 
  store_packet(From, To, Packet) ->
      Type = xml:get_tag_attr_s("type", Packet),
      if
***************
*** 96,101 ****
--- 120,127 ----
  	(Type /= "headline") ->
  	    case check_event(From, To, Packet) of
  		true ->
+ 		case check_alias_forward(From, To, Packet) of
+ 		true ->
  		    #jid{luser = LUser, lserver = LServer} = To,
  		    TimeStamp = now(),
  		    {xmlelement, _Name, _Attrs, Els} = Packet,
***************
*** 109,114 ****
--- 135,144 ----
  				     packet = Packet},
  		    stop;
  		_ ->
+ 			offline_msg
+ 			end,
+ 			stop;
+ 		_ ->
  		    ok
  	    end;
  	true ->
